roadmap/implementers-guide/src/node/utility/chain-api.md

```diff
@@ -0,0 +1,21 @@
+# Chain API
+
+The Chain API subsystem is responsible for providing a single point of access to chain state data via a set of pre-determined queries.
+
+## Protocol
+
+Input: [`ChainApiMessage`](../../types/overseer-protocol.md#chain-api-message)
+
+Output: None
+
+## Functionality
+
+On receipt of `ChainApiMessage`, answer the request and provide the response to the side-channel embedded within the request.
+
+Currently, the following requests are supported:
+* Block hash to number
+* Finalized block number to hash
+* Last finalized block number
+* Ancestors
+
+> TODO Do we need caching at this layer or is it already done on a lower level?
```

Comment:
The lower level does handle caching, so we generally shouldn't need it. I doubt this is our lowest-hanging fruit, though.